ixwebsocket/IXWebSocketTransport.cpp

```diff
@@ -300,24 +309,12 @@ namespace ix
                 }
                 else if (ret <= 0)
                 {
-                    _socket->close();
-
                     // if there are received data pending to be processed, then delay the abnormal closure
                     // to after dispatch (other close code/reason could be read from the buffer)
-                    if (_rxbuf.size() > 0) 
-                    {
-                        _treatAbnormalCloseAfterDispatch = true;
-
-                        setReadyState(CLOSING);
-                    }
-                    // no received data pending processing, so we can close directly
-                    else
-                    {   
-                        _treatAbnormalCloseAfterDispatch = false;
-                        internalClose(kAbnormalCloseCode, kAbnormalCloseMessage, 0, true);
-                    }
+                    
+                    _socket->close();
 
-                    break;
```

Comment:
I've started to use enum class in some places, they are more elegant that straight enum. We don't need to change this but just fyi.